pkg/operator/controller/dns/controller.go

```diff
@@ -573,14 +576,23 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 			log.Info("using fake DNS provider because cluster's ControlPlaneTopology is External")
 			return &dns.FakeProvider{}, nil
 		}
+
 		var err error
-		cisInstanceCRN := platformStatus.IBMCloud.CISInstanceCRN
-		if cisInstanceCRN == "" {
-			return nil, fmt.Errorf("missing cis instance crn")
-		}
-		dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, cisInstanceCRN, userAgent)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create IBM DNS manager: %v", err)
+		if platformStatus.IBMCloud.CISInstanceCRN != "" {
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, platformStatus.IBMCloud.CISInstanceCRN, userAgent, true)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create ibm cis dns provider: %v", err)
```

Comment:
Looks like you're changing "IBM CIS DNS" to upper-case elsewhere, right?  Anyway, might as well change to using `%w` while you're updating this code.  
```suggestion
				return nil, fmt.Errorf("failed to create IBM CIS DNS provider: %w", err)
```
Actually, `getIbmDNSProvider` already wraps whatever error it returns with "failed to initialize IBM CIS DNS provider", so is it necessary to wrap the error here?  Might as well just return it as-is, no?
```suggestion
				return nil, err
```

Reply:
Correct, changed it.